File: 0429.N-ary_tree_level_order_traversal/solution.java
import java.util.List;

class Node {
    public int val;
    public List<Node> children;

    public Node() {}

    public Node(int _val,List<Node> _children) {
        val = _val;
        children = _children;
    }
}

class Solution {
    public List<List<Integer>> levelOrder(Node root) {
        List<List<Integer>> ans = new ArrayList<>();
        if (root == null) return ans;
        Queue<Node> nodes = new LinkedList<>();
        nodes.add(root);
        while (!nodes.isEmpty()){
            int size = nodes.size();
            List<Integer> temp = new ArrayList<>();
            for (int i=0; i< size; i++){
                temp.add(nodes.peek().val);
                nodes.addAll(nodes.poll().children);
            }
            ans.add(temp);
        }
        return ans;
    }
}

File: 0429.N-ary_tree_level_order_traversal/solution.py
class Node:
    def __init__(self, val, children):
        self.val = val
        self.children = children


class Solution:
    def level_order(self, root: "Node") -> List[List[int]]:
        ans, nodes = [], [root]
        while nodes:
            temp, temp_nodes = [], []
            for node in nodes:
                temp.append(node.val)
                temp_nodes.extend(node.children)
            ans.append(temp)
            nodes = temp_nodes
        return ans
